author	Marc Deslauriers <marc.deslauriers@ubuntu.com>	2024-06-19 10:33:00 -0400
committer	git-ubuntu importer <ubuntu-devel-discuss@lists.ubuntu.com>	2024-06-26 12:22:56 +0000
commit	7584287c1cf59926252197badedde2cbc08e084c (patch)
tree	246e4fa465245f04c53f82cfb8cfeda7ea843db4
parent	7128299adb87ba73094732751d96621648db1bce (diff)
[PATCH] Use safe_load and safe_load_file for .rdoc_options
Gbp-Pq: CVE-2024-27281-2.patch.
Diffstat
-rw-r--r--	lib/rdoc/rdoc.rb	3
-rw-r--r--	test/rdoc/test_rdoc_options.rb	6
2 files changed, 5 insertions, 4 deletions
diff --git a/lib/rdoc/rdoc.rb b/lib/rdoc/rdoc.rb
index 605172ae..f6105c68 100644
--- a/lib/rdoc/rdoc.rb
+++ b/lib/rdoc/rdoc.rb
@@ -156,8 +156,9 @@ class RDoc::RDoc
     RDoc.load_yaml

     begin
-      options = YAML.load_file '.rdoc_options'
+      options = YAML.safe_load_file '.rdoc_options', permitted_classes: [RDoc::Options, Symbol]
     rescue Psych::SyntaxError
+      raise RDoc::Error, "#{options_file} is not a valid rdoc options file"
     end

     raise RDoc::Error, "#{options_file} is not a valid rdoc options file" unless
diff --git a/test/rdoc/test_rdoc_options.rb b/test/rdoc/test_rdoc_options.rb
index 140c4afc..f547f5bf 100644
--- a/test/rdoc/test_rdoc_options.rb
+++ b/test/rdoc/test_rdoc_options.rb
@@ -145,7 +145,7 @@ class TestRDocOptions < RDoc::TestCase

     @options.encoding = Encoding::IBM437

-    options = YAML.load YAML.dump @options
+    options = YAML.safe_load(YAML.dump(@options), permitted_classes: [RDoc::Options, Symbol])

     assert_equal Encoding::IBM437, options.encoding
   end
@@ -161,7 +161,7 @@ rdoc_include:
 - /etc
     YAML

-    options = YAML.load yaml
+    options = YAML.safe_load(yaml, permitted_classes: [RDoc::Options, Symbol])

     assert_empty options.rdoc_include
     assert_empty options.static_path
@@ -749,7 +749,7 @@ rdoc_include:

       assert File.exist? '.rdoc_options'

-      assert_equal @options, YAML.load(File.read('.rdoc_options'))
+      assert_equal @options, YAML.safe_load(File.read('.rdoc_options'), permitted_classes: [RDoc::Options, Symbol])
     end
   end

